src/librustc/hir/mod.rs

```diff
@@ -2060,10 +2060,8 @@ pub struct InlineAsmOutput {
     pub span: Span,
 }
 
-// NOTE(eddyb) This is used within MIR as well, so unlike the rest of the HIR,
-// it needs to be `Clone` and use plain `Vec<T>` instead of `HirVec<T>`.
 #[derive(Clone, RustcEncodable, RustcDecodable, Debug, HashStable)]
-pub struct InlineAsm {
+pub struct InlineAsmInner {
```

Comment:
I think I'd prefer this to stay `InlineAsm` and name the other struct `InlineAsmInvoc` or sth.

Reply:
This would be inconsistent with `ast::InlineAsm` and `mir::InlineAsm` which represents the whole thing -- `InlineAsmInner` only exists here due to the `Clone` and `TypeFoldable` kerfuffle -- ideally it shouldn't exist at all.